src/common/components/Table/TableComponent.tsx

```diff
@@ -0,0 +1,119 @@
+import { DataGrid } from '@mui/x-data-grid';
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '@mui/material/IconButton';
+import useTheme from '@mui/material/styles/useTheme';
+import DeleteIcon from '@mui/icons-material/Delete';
+import EditIcon from '@mui/icons-material/Edit';
+import { TableComponentProps } from './interfaces';
+import { useTranslation } from 'react-i18next';
+
+const TableComponent: React.FC<TableComponentProps> = ({
+  dataObject,
+  tableSize,
+  onDelete,
+  showDeleteButton = false,
+  showEditButton = false,
+  onEdit,
+}) => {
+  const theme = useTheme();
+  const MyColor = theme.palette.primary.main;
+  const isLarge = tableSize === 'large';
+  const tableHeight = isLarge ? '300px' : '200px';
+  const tableWidth = isLarge ? '85%' : '60%';
+  const [pageSize, setPageSize] = useState<number>(5);
+  const { t } = useTranslation();
+
+  const columns = dataObject.headers.map((header) => ({
+    field: header.key,
+    headerName: header.label,
+    flex: 1,
+    renderCell: header.renderCell,
+  }));
+
+  if (isLarge && (showDeleteButton || showEditButton)) {
+    columns.push({
+      field: 'actions',
+      headerName: '',
+      flex: 0.5,
+      renderCell: (params: any) => (
+        <div style={{ display: 'flex', gap: '8px' }}>
+          {showEditButton && (
+            <IconButton
+              onClick={() => onEdit?.(params.row)}
+              aria-label='edit'
+              sx={{ color: MyColor }}
+            >
+              <EditIcon />
+            </IconButton>
+          )}
+          {showDeleteButton && (
+            <IconButton
+              onClick={() => onDelete?.(params.row.id)}
+              aria-label='delete'
+              sx={{ color: MyColor }}
+            >
+              <DeleteIcon />
+            </IconButton>
+          )}
+        </div>
+      ),
+    });
+  }
+
+  return (
+    <div
+      style={{
+        height: tableHeight,
+        width: tableWidth,
+        marginTop: '20px',
+        marginRight: tableSize === 'large' ? 'auto' : '0',
+        display: 'flex',
+      }}
+    >
+      {dataObject.rows.length === 0 ? (
+        <div style={{ textAlign: 'center', padding: '20px', color: '#888' }}>
+          {t('Nothing to display, please enter data.')}{' '}
```

Comment:
זה עובד בצורה כזאת?
בדרך כלל צריך קידומת כלשהי, למשל common.'nothing to display'